multiarea_model/analysis_helpers.py

```diff
@@ -668,8 +668,11 @@ def synchrony(data_array, num_neur, t_min, t_max, resolution=1.0):
     spike_count_histogramm = pop_rate_time_series(
         data_array, num_neur, t_min, t_max, resolution=resolution)
     mean = np.mean(spike_count_histogramm)
-    variance = np.var(spike_count_histogramm)
-    synchrony = variance / mean
+    std_dev = np.std(spike_count_histogramm)
+    if std_dev + mean != std_dev:  # check for vanishing mean
```

Comment:
Maybe I am a bit pedantic here, but addition is limited by floating point precision. E.g.
```
In [1]: 0.1 + 0.2                                                               
Out[1]: 0.30000000000000004

In [2]: 0.1 + 0.2 == 0.3                                                        
Out[2]: False

```
I don't know if these floating point errors can also happen when you add 0. What about?
```
try:
    synchrony = std_dev / mean
except ZeroDivisionError:
    synchrony = np.inf
```

Reply:
Thank you @AlexVanMeegen and @jarsi,
> I don't know if these floating point errors can also happen when you add 0
Just for information, actually that kind of check is motivated exactly by the fact that addition is limited by floating point precision. As I said, it's a standard trick, you can find it e.g. in "Numerical recipes in C. The Art of Scientific Computing". a + b should always be == a if b is 0 or smaller than the floating point precision of a. That said, for sure your solution is much more elegant!

Reply:
Also think your solution, @jarsi, is cleaner. Thanks for being pendantic :nerd_face: